bletl/growth.py

```diff
@@ -500,3 +456,69 @@ def fit_mu_t(
         result.sample(draws=mcmc_samples)
 
     return result
+
+
+def detect_switchpoints(
+    switchpoint_prob: float,
+    t_data: typing.Sequence[float],
+    pmodel: pm.Model,
+    theta_map: typing.Dict[str, numpy.ndarray],
+) -> typing.Dict[float, str]:
+    """Helper function to detect switchpoints from a fitted random walk.
+
+    Parameters
+    ----------
+    switchpoint_prob
+        Probability threshold for detecting switchpoints.
+        Random walk innovations with a prior probability less than this
+        will be classified as switchpoints.
+    t_data
+        Time values corresponding to the random walk steps.
+    pmodel
+        The PyMC model containing `"mu_t*"` random walks.
+    theta_map
+        MAP estimate of the model.
+
+    Returns
+    -------
+    switchpoints
+        Dictionary of switchpoints with
+        keys being the time point and
+        values `"detected"`.
+    """
+    # first CDF values at all mu_t elements
+    cdf_evals = []
+    for rvname in sorted(theta_map.keys()):
+        if rvname not in pmodel.named_vars:
+            continue
+        # The random walk may be split in multiple segments.
+        # We can identify a segment from the RVOp type that created it.
+        rv = pmodel[rvname]
+        if rv.owner is None:
+            continue
+        if isinstance(rv.owner.op, pm.RandomWalk.rv_type):
+            # Get a handle on the innovation dist
+            innov_dist = rv.owner.inputs[1]
+            # and calculate the innovations from the MAP estimate
+            innov = numpy.diff(theta_map[rvname])
+            # so we can calculate the logcdf and cdf evaluations.
+            logcdfs = pm.logcdf(innov_dist, innov).eval()
+            # Prepend a 0.5 because the init dist does not count into the diff.
+            cdf_evals += [0.5, *numpy.exp(logcdfs)]
```

Comment:
Why a 0.5 here?

Reply:
0.5 is the CDF value when falling exactly onto the prior. A `0` would be below that threshold a few lines down..

Reply:
I updated these comments to be more explanatory